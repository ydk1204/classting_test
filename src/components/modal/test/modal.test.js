import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import renderer from 'react-test-renderer';
import Modal from '../modal';

const setIsModal = false;

describe('Modal', () => {
  it('renders', () => {
    const component = renderer.create(<Modal setIsModal={setIsModal} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});