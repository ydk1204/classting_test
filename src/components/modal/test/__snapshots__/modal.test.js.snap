// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders 1`] = `
<div
  className="w-screen h-full flex justify-center items-center fixed top-0 left-0 bg-black bg-opacity-70 z-30 md:h-screen"
>
  <div
    className="relative flex flex-col w-screen h-screen bg-white top-0 rounded-none overflow-hidden md:flex-row md:w-[90vw] md:h-[70vh] md:rounded-lg xl:h-[90vh]"
  >
    <div
      className="basis-auto h-[40%] md:basis-1/2 md:h-full"
    >
      <img
        alt="퀴즈 페이지"
        className="w-full h-full object-fit"
        src="./img/classting.jpeg"
      />
    </div>
    <div
      className="basis-auto p-5 w-full h-full overflow-auto md:p-20 md:basis-1/2"
    >
      <button
        className="absolute top-8 right-8 text-2xl w-14 h-14 rounded-lg bg-white hover:bg-black hover:text-white transition duration-200 ease-in-out"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-xmark "
          data-icon="xmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M310.6 361.4c12.5 12.5 12.5 32.75 0 45.25C304.4 412.9 296.2 416 288 416s-16.38-3.125-22.62-9.375L160 301.3L54.63 406.6C48.38 412.9 40.19 416 32 416S15.63 412.9 9.375 406.6c-12.5-12.5-12.5-32.75 0-45.25l105.4-105.4L9.375 150.6c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0L160 210.8l105.4-105.4c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25l-105.4 105.4L310.6 361.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
      <div
        className="flex flex-col items-start w-full h-full"
      >
        <h1
          className="text-4xl"
        >
          퀴즈 풀이 방식
        </h1>
        <div
          className="mt-16 whitespace-pre-line text-lg py-5"
        >
          <ul
            className="modal-ul"
          >
            <li>
              1. 퀴즈 풀기 버튼 클릭 시 퀴즈를 풀 수 있는 페이지로 이동합니다.
            </li>
            <li>
              2. 사용자는 문항에 대한 답안을 4개의 보기 중 하나를 선택할 수 있습니다.
            </li>
            <li>
              3. 답안이 맞았는지 틀렸는지 바로 알 수 있습니다.
            </li>
            <li>
              4. 답안을 선택하면 다음 문항을 볼 수 있습니다.
            </li>
            <li>
              5. 모든 문항을 다 풀면 사용자는 정답/오답 개수를 볼 수 있습니다.
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
